rln/src/merkle_tree.rs

```diff
@@ -0,0 +1,377 @@
+// Implementation adapted from https://github.com/worldcoin/semaphore-rs/blob/d462a4372f1fd9c27610f2acfe4841fab1d396aa/src/merkle_tree.rs
+//! Implements basic binary Merkle trees
+//!
+//! # To do
+//!
+//! * Disk based storage backend (using mmaped files should be easy)
+
```

Comment:
If I understand it correctly this file is basically same with minor addition to keep track of capacity. Perhaps this can be added as a comment here? So easier to keep up to date.

Reply:
Addressed in https://github.com/vacp2p/zerokit/pull/22/commits/28b96a60a631beaebf8386ac5f727322c759d440